### ISSUE TITLE
Application crashes when I try to get password

### ISSUE THREAD

**Issue by devalentino** [mssun/passforios#264]
I installed application, configured ssh key and gpg keys, pulled passwords but was not able to see password in app. When I press password name (for example trello), application exits. Here is screen recording https://drive.google.com/file/d/1TTlODXqyoT8Cu8aNIz9AjGip9mPPJBgB/view

**Comment by yishilin14** [comment]
It seems like decrypting the entry crashes the app. Are you able to add a new entry?

What kind of gpg keys are you using? Do you mind posting a pair of new keys here or sending them to 
developer@passforios.mssun.me so I can reproduce the issue?

**Comment by devalentino** [comment]
I successfully created new key, pushed to git repo and red on my laptop. I use 2048 gpg key.
Not sure it is safe to share my gpg keys

**Comment by yishilin14** [comment]
Because you can use Pass in your laptop, I guess your keys are not supported somehow by Pass (by the upstreaming libraries Pass uses).

It is not safe to share keys being used. You could generate a pair of new keys with fake name and email.

**Comment by devalentino** [comment]
I regenerated test gpg key, created test git repository and added password on my laptop. Was able to see it in app. In original case I keep my gpg keys on yubikey, maybe export doesn't work properly from it?
Will try to generate new gpg keys for my yubikey

**Comment by PhotonQuantum** [comment]
I've met with the same problem. My key pair is rsa 4096 encrypted, and has a 4kb image. Original pass works fine but pass4ios crashes after I edit/create anything. If needed, I may attach a dummy key pair next week.

**Comment by yishilin14** [comment]
@PhotonQuantum You are very welcomed to provide a minimal example (e.g, a dummy key pair and detailed steps) to reproduce the crash.

**Comment by PhotonQuantum** [comment]
@yishilin14 Strangely enough, when I use the TestFlight version, the problem just disappears. Or maybe it's just a random behavior? I'll post more information here as long as I reproduce the crash again.

**Comment by PhotonQuantum** [comment]
Now I figured out what's happened.
I have two encryption keys. The first one is on my yubikey, and the second one is saved on my computer.
When I was exporting my second key, I forgot to append '!' to my keyid, resulting in a full keyring export, and Pass4ios then used the yubikey one to encrypt messages.
Encrypting was fine because the its pubkey part was complete, however decryption failed because the private part needed was partially saved on my yubikey.
So basically there's no flaw in the app.
@devalentino Did you save your encryption subkey to your yubikey with 'keytocard', and exported this subkey to your phone later? Then your exported key is corrupted. GPG saves essential parts of your subkey to yubikey, then it deletes them from your computer, leaving you a partial subkey.

**Comment by mssun** [comment]
Seems like a problem related to the PGP library. Thanks to @ZortacDev, we have changed to GopenPGP which may potentially resolves many existing PGP encryption algorithm issues. The latest version 0.6.0 (20) is in the TestFlight channel. Please try the latest TestFlight and report any issues related to the PGP encryption/decryption. Thanks.

**Comment by mssun** [comment]
I'm closing this issue since we have released a new version in App Store to (to some extent) solve the PGP issues. Feel free to reopen it if you believe the issue is still valid.
